tests/program_analysis/CAST2GrFN/test_cast_json.py

```diff
@@ -0,0 +1,164 @@
+import json
+import pytest
+
+from automates.program_analysis.CAST2GrFN.model.cast import (
+    AstNode,
+    Assignment,
+    Attribute,
+    BinaryOp,
+    BinaryOperator,
+    Call,
+    ClassDef,
+    Dict,
+    Expr,
+    FunctionDef,
+    List,
+    Loop,
+    ModelBreak,
+    ModelContinue,
+    ModelIf,
+    ModelReturn,
+    Module,
+    Name,
+    Number,
+    Set,
+    String,
+    Subscript,
+    UnaryOp,
+    UnaryOperator,
+    VarType,
+    Var,
+)
+
+from automates.program_analysis.CAST2GrFN.cast import CAST, CASTJsonException
+
+
+def get_file_str(filepath):
```

Comment:
@dpdicken this design pattern is unnecessary for JSON files. Please use the following instead:

```python
import json

my_data = json.load(open("<filepath>", "r"))
```

Reply:
I know that this is an option, however the tests are parsing the CAST from a json string and also comparing the json string from the file to the json string generated by the "to_json_str" method. AFAIK json.load returns an object if I am not mistaken. Perhaps they should be comparing objects instead and this change would make sense.

Reply:
> Perhaps they should be comparing objects instead and this change would make sense.
I would definitely prefer for us to compare objects. That will likely provide us with richer diffs if/when anything does differ between the objects we are comparing. Is that not the case? Additionally, if we need more packages to make the object comparison easier then we can talk about including those in AutoMATES.